xapian-letor/bin/xapian-rank.cc

```diff
@@ -101,7 +102,11 @@ try {
 		have_database = true;
 		break;
 	    case 'm':
-		msize = atoi(optarg);
+		unsigned int size;
+		if (!parse_unsigned(optarg, size)) {
+		    throw "Mset size must be >= 0";
+		}
+		msize = size;
```

Comment:
The comments from `xapian-prepare-trainingfile.cc` apply here too.

Reply:
fixed in a fixup commit.